src/xava.c

```diff
@@ -224,6 +224,22 @@ as of 0.4.0 all options are specified in config file, see in '/home/username/.co
         thing.xava = &xava;
         xavaIONotifyAddWatch(&thing);
 
+        // attach pywal colors file to the IONotify thing
+        static const char b[] = "/.cache/wal/colors";
+        int lena = strlen(getenv("HOME"));
+        int lenb = strlen(b);
+        char *filename = malloc(lena+lenb+1);
+        // copy & concat (including string termination)
+        memcpy(filename,getenv("HOME"),lena);
+        memcpy(filename+lena,b,lenb+1);   
+        struct xava_ionotify_watch_setup pywal;
+        pywal.xava_ionotify_func = &handle_ionotify_call;
```

Comment:
Potential improvment: Since these changes are mostly colour related, we don't really need to reload the entire visualizer, instead we can just apply colours during runtime using a seperate ``handle_ionotify_call`` or make ``handle_ionotify_call`` handle logic differently based on the ID parameter (we can use it to find out what triggered the callback). 

Something like: ``XAVA_IONOTIFY_CALLBACK_MAIN`` and ``XAVA_IONOTIFY_CALLBACK_PYWAL`` just to make it clear for other people reading the code.